milli/src/score_details.rs

```diff
@@ -0,0 +1,295 @@
+use serde::Serialize;
+
+use crate::distance_between_two_points;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum ScoreDetails {
+    Words(Words),
+    Typo(Typo),
+    Proximity(Rank),
+    Fid(Rank),
+    Position(Rank),
+    ExactAttribute(ExactAttribute),
+    Exactness(Rank),
+    Sort(Sort),
+    GeoSort(GeoSort),
+}
+
+impl ScoreDetails {
+    pub fn local_score(&self) -> Option<f64> {
+        self.rank().map(Rank::local_score)
+    }
+
+    pub fn rank(&self) -> Option<Rank> {
+        match self {
```

Comment:
I hesitated for a long time to make it so the "highest" rank is 0, and the worst rank is "max rank", so that they're ordinals, like in a race. But then I kinda want to start them at 1 instead of 0 😬 

Would maybe `Grade` or `Mark` work better than `Rank` for you?

Today the main distinction between rank and score is that the score is normalized by dividing the rank by its max rank.

Wording is important, so I'm very open to make the necessary changes here, I'm just a bit unsure what these are :-)